Sources/Sentry/Public/SentrySDK.h

```diff
@@ -372,51 +372,43 @@ SENTRY_NO_INIT
 /**
  * Start a new continuous profiling session if one is not already running.
  * @warning Continuous profiling mode is experimental and may still contain bugs.
- * @warning This property is deprecated and will be removed in a future version of the SDK. See
- * @c startProfileSession .
- * @note Unlike trace-based profiling, continuous profiling does not take into account @c
- * SentryOptions.profilesSampleRate ; a call to this method will always start a profile if one is
- * not already running. This includes app launch profiles configured with @c
- * SentryOptions.enableAppLaunchProfiling .
- * @seealso https://docs.sentry.io/platforms/apple/guides/ios/profiling/#continuous-profiling
- */
-+ (void)startProfiler;
-
-/**
- * Start a new continuous profile session.
- * @warning Continuous profiling mode is experimental and may still contain bugs.
- * @note Does nothing if the profiling session is sampled with respect
- * @c SentryOptions.profileSessionSampleRate and the profiler is already running.
- * @note Does nothing if the profile session is not sampled.
- * @note Does nothing if @c SentryOptions.profileLifecycle is set to @c trace . In this scenario,
- * the profiler is automatically started and stopped depending on whether there is an active sampled
- * span so it is not permitted to manually start profiling.
+ * @note Unlike trace-based profiling, continuous profiling does not take into account
+ * @c SentryOptions.profilesSampleRate or @c SentryOptions.profilesSampler . If either of those
+ * options are set, this method does nothing. @b Warning: Because those options are deprecated, this
+ * behavior is deprecated and will change in a future version.
+ * @note Taking into account the above note, if @c SentryOptions.configureProfiling is not set,
+ * calls to this method will always start a profile if one is not already running. This includes app
+ * launch profiles configured with @c SentryOptions.enableAppLaunchProfiling . If you need sampling,
+ * you must calculate your own sampling decisions to call this method or enable app launch
+ * profiling. @b Warning: this behavior relies on deprecated options and will change in a future
+ * version. In the future, there will be internal sampling and the default sample rate will be 0.
+ * @note If neither @c SentryOptions.profilesSampleRate nor @c SentryOptions.profilesSampler are
+ * set, and @c SentryOptions.configureProfiling is set, this method does nothing if the profiling
+ * session is not sampled with respect to @c SentryOptions.profileSessionSampleRate , (which
+ * defaults to a sample rate of 0), or if it is sampled but the profiler is already running.
+ * @note If neither @c SentryOptions.profilesSampleRate nor @c SentryOptions.profilesSampler are
+ * set, and @c SentryOptions.configureProfiling is set, this method does nothing if
```

Comment:
This documentation looks like parts of this are duplicated (e.g. the `@warning`) and there's other stuff that doesn't apply: `In the future, there will be internal sampling and the default sample rate will be 0.` -- we do have internal sampling now

Reply:
This part is tricky. I'm trying to warn people using v1 of continuous profiling, where we had no sampling facility, that in SDK version 9.0.0, the behavior of this method will switch out from under them, because it will go from having no sampling (continuous profiling v1) to having a default sample rate of 0 (continuous profiling v2) so if they don't configure the profile session sample rate when they upgrade, any calls to this method they already had will just go dark.

Reply:
IMO we should put that in the changelog/release notes - having this here for new customers who don't have the context will be more confusing. We're OK with breaking the prior behavior because it was a beta feature (and we weren't billing for usage).

Reply:
Good point 👍🏻 That'll simplify these a lot.